include/sound/sof/ext_manifest.h

```diff
@@ -61,6 +61,18 @@ enum sof_ext_man_elem_type {
 	SOF_EXT_MAN_ELEM_WINDOW			= SOF_IPC_EXT_WINDOW,
 	SOF_EXT_MAN_ELEM_CC_VERSION		= SOF_IPC_EXT_CC_INFO,
 	SOF_EXT_MAN_ELEM_DBG_ABI		= SOF_IPC_EXT_USER_ABI_INFO,
+	SOF_EXT_MAN_ELEM_CONFIG_DATA		= SOF_IPC_EXT_CONFIG_DATA,
+};
+
+/* Extended manifest elements identificators */
+enum sof_config_elem_type {
+	SOF_EXT_MAN_CONFIG_CAVS_LPRO	= 1,
+	SOF_EXT_MAN_CONFIG_LAST_ELEM,	/*< keep it at the end of enum list */
+};
+
+struct sof_config_elem {
+	uint32_t token;
+	uint32_t value;
```

Comment:
doesn't this struct have to be packed?

Reply:
Thank you for pointing this. I blindly copied this structure from SOF firmware change. https://github.com/thesofproject/sof/pull/2893
All the structures need to be packed?  I should ask this to sof's PR also.

Reply:
@lyakh My understanding, __pack is not MUST. As long as we use same __pack(or not) between linux and sof firmware, that should be fine, right? What do you think?

Reply:
@fredoh9 no, that isn't how I understand it. `__packed` must be used in certain well defined cases. Basically everywhere where we have to guarantee the binary layout of data in memory, because without that attribute the compiler has some liberty about arranging data. So, if you pass a data structure over a network or over a bus you must guarantee, that all parties pack and unpack the structure in the same way.
This is definitely one of such cases - this struct is a part of the manifest, i.e. it's generated by one partie, then parsed by other parties. I have no doubts that this struct must be packed, my only doubt was whether it has to be declared as `__packed` in its own definition, because for inclusion in the manifest it's embedded in `struct sof_ext_man_config_data` which is already `__packed`. But I still think you have to declare this one `__packed` too, because actually I don't think the packing of embedding struct affects that of the embedded struct, because that would change the layout of the embedded struct depended on where it's embedded... So, I do think you have to declare it `__packed` too.

Reply:
@fredoh9 @lyakh yes, it should be __packed too, thanks. I missed this because two uint32_t members are perfectly aligned right now :)
FW PR updated

Reply:
@lyakh thank you for detailed description. I had very narrow understanding till now.
@abonislawski thank you for update, I will update mine too.